tests/location/UserLocation.test.ts

```diff
@@ -0,0 +1,90 @@
+import { LocationAccuracy } from "expo-location"
+import {
+  expoTrackUserLocation,
+  LocationCoordinatesMocks,
+  UserLocationTrackingAccurracy
+} from "@lib/location"
+
+const testAccuracy = "precise"
+
+describe("UserLocation tests", () => {
+  describe("ExpoUserLocationTracking tests", () => {
+    test("translates accurracy to proper expo accurracy when making tracking request", () => {
+      expectAccuracyConversion("approximate-low", LocationAccuracy.Low)
+      expectAccuracyConversion("approximate-medium", LocationAccuracy.Balanced)
+      expectAccuracyConversion("precise", LocationAccuracy.Highest)
+    })
+
+    test("translates expo location update into TrackedLocation", () => {
+      const callback = jest.fn()
+      const track = jest.fn().mockImplementation((_, trackFn) => {
+        trackFn({
+          coords: LocationCoordinatesMocks.Paris,
+          timestamp: 1000
+        })
+        return Promise.resolve()
+      })
+
+      expoTrackUserLocation(testAccuracy, callback, track)
+      expect(callback).toHaveBeenCalledWith({
+        status: "success",
+        location: {
+          coordinates: LocationCoordinatesMocks.Paris,
+          trackingDate: new Date(1)
+        }
+      })
+    })
+
+    it("sends an error update when expo tracker fails", async () => {
+      const callback = jest.fn()
+      const track = jest.fn()
+
+      // NB: Expo makes their API asynchronous, however since we're in
+      // a syncronous context we use a promise to ensure that our callback
+      // actually consumes the error event before the test ends.
+      const errorPromise = new Promise<void>((resolve) => {
+        track.mockImplementation(() => {
+          resolve()
+          return Promise.reject(new Error())
+        })
+      })
+
+      expoTrackUserLocation(testAccuracy, callback, track)
+      await errorPromise
+      expect(callback).toHaveBeenCalledWith({ status: "error" })
+    })
+
+    it("unsubscribes from expo tracking when unsub invoked", async () => {
+      const track = jest.fn()
+      const unsubFromExpo = jest.fn()
+
+      // NB: The unsub function is in a "fire-and-forget" style, which means
+      // we can't tell when it will actually unsubscribe from expo (since
+      // expo's tracker returns a promise). This promise ensures that we give
+      // a chance for the tracker to resolve before asserting.
+      const trackPromise = new Promise<void>((resolve) => {
+        track.mockImplementation(() => {
+          resolve()
+          return Promise.resolve({ remove: unsubFromExpo })
+        })
+      })
+
+      const unsub = expoTrackUserLocation(testAccuracy, jest.fn(), track)
```

Comment:
We can also leave this out since there aren't really edge cases here, it's just checking react native's useEffect cleanup. Also people will likely not use expoTrackUserLocation directly so manual unsubbing isn't something to worry about.

Reply:
You're right about the case where inside a `useEffect` it would be useless to test (since we get the cleanup logic for free). However, in this case the test is testing for what the unsubscribe function _actually_ does since this test isn't inside a react component. In this case, it simply means, once we have subscribed to the tracker (which may not be the case when `unsub` is called, and thus we have to wait), do we actually unsubscribe.